src/process_utility.c

```diff
@@ -472,10 +472,13 @@ process_alter_foreign_server(ProcessUtilityArgs *args)
 {
 	AlterForeignServerStmt *stmt = (AlterForeignServerStmt *) args->parsetree;
 
-	if (block_on_foreign_server(stmt->servername))
+	if (stmt->has_version)
```

Comment:
I wonder if we should move foreign server related code and blocking logic to the dist ddl instead

Reply:
I thought about it but didn't see a need for it at this point.